core/io/resource_saver.cpp

```diff
@@ -120,9 +120,8 @@ Error ResourceSaver::save(const Ref<Resource> &p_resource, const String &p_path,
 
 		String local_path = ProjectSettings::get_singleton()->localize_path(path);
 
-		Ref<Resource> rwcopy = p_resource;
```

Comment:
This probably was done on the false assumption that `const Ref<Resource> &p_resource` can't be changed (or it was this way at some point, since it dates back to `GODOT IS OPEN SOURCE`).